src/pudl/transform/ferc714.py

```diff
@@ -293,6 +317,26 @@ def _assign_respondent_id_ferc714(
     return df
 
 
+def _fillna_respondent_id_ferc714_source(
+    df: pd.DataFrame, source: Literal["csv", "xbrl"]
+) -> pd.DataFrame:
+    """Fill missing CSV or XBRL respondent id."""
+    respondent_map_ferc714 = pd.read_csv(
+        importlib.resources.files("pudl.package_data.glue")
+        / "respondent_id_ferc714.csv"
+    ).convert_dtypes()
+    # use the source utility ID column to get a unique map and for merging
+    resp_id_col = f"respondent_id_ferc714_{source}"
+    resp_map_series = respondent_map_ferc714.dropna(subset=[resp_id_col]).set_index(
+        "respondent_id_ferc714"
+    )[resp_id_col]
+
+    df[resp_id_col] = df[resp_id_col].fillna(
+        df["respondent_id_ferc714"].map(resp_map_series)
+    )
+    return df
```

Comment:
I'm not sure I understand how this is working. The `_assign_respondent_id_ferc714` function maps the `respondent_id_ferc714` column and then this column appears to work backwards from that to map on missing `respondent_id_ferc714_source` values. How is that possible? If there is no `respondent_id_ferc714_source` to begin with how can we map a `respondent_id_ferc714` value onto it?

Reply:
Also not sure I understand why this is important when we drop the respondent id source columns anyways?

Reply:
Zane gave me verbal feedback about supporting my leaning towards keeping the source IDs at least in the long hourly table.

This step really only makes sense to run *after* we have added in the `respondent_id_ferc714`. so everything has a pudl-derived id. but also in the context of the hourly table, doing this would enable users to come to the data with the source ID that they are the most familiar with and get the full time series (so if they knew the CSV-based respondent ID and selected records based on that they'd get the all of the years for that respondent including XBRL years if the respondent is mapped the an xbrl respondent.